h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -292,12 +292,15 @@ to a map (possibly the metadata map) -
         Object o = config.get("compress");
         this.compressionLevel = o == null ? 0 : (Integer) o;
         String fileName = (String) config.get("fileName");
-        o = config.get("pageSplitSize");
-        if (o == null) {
-            pageSplitSize = fileName == null ? 4 * 1024 : 16 * 1024;
-        } else {
-            pageSplitSize = (Integer) o;
+        fileStore = (FileStore) config.get("fileStore");
+        fileStoreIsProvided = fileStore != null;
+        if(fileStore == null && fileName != null) {
+            fileStore = new FileStore();
         }
+        o = config.get("pageSplitSize");
+        pageSplitSize = o != null         ? (Integer) o :
+                        fileStore == null ? 48 :
+                                            16 * 1024;
```

Comment:
While I amd definitely not religious on such matters (any good thing can be used and can be abused), code readability is somewhat **subjective** thing. Unfortunately, the more verbose alternative to ternary operation stays more verbose at the bytecode level as well, even with javac 1.8 in non-debugging mode. And that's something **tangible**, which I generally wouldn't trade for "code readability". In this non-performance-critical code, it's a moot point, of course. Will change...